macros/DekiScript-Page.ejs

```diff
@@ -1,225 +0,0 @@
-    subpagesExpand: function (path, depth, self) {
-        var mdn = require_macro('MDN:Common');
-        var url = build_api_url((path ? path : env.url) + '$children?expand');
-        var depth_check = parseInt(depth);
-        if (depth_check >= 0) {
-            url += '&depth=' + depth_check;
-        }
-        var subpages = mdn.fetchJSONResource(url);
-        var result = [];
-        if (subpages != null) {
-            if (!self) {
-                result = subpages.subpages || [];
-            } else {
-                result = [subpages];
-            }
-        }
-        return result;
-    },
-
-    // If reverse is non-zero, the sort is backward
-    subPagesSort: function (pages, reverse) {
-        if (reverse == 0) {
-            pages.sort(alphanumForward);
-        } else {
-            pages.sort(alphanumBackward);
-        }
-
-        return pages;
-
-        function chunkify(t) {
-        	var tz = [], x = 0, y = -1, n = 0, i, j;
-
-			while (i = (j = t.charAt(x++)).charCodeAt(0)) {
-                var m = (i == 46 || (i >=48 && i <= 57));
-				if (m !== n) {
-					tz[++y] = "";
-	    			n = m;
-				}
-				tz[y] += j;
-			}
-			return tz;
-		}
-
-        function alphanumForward(a, b) {
-
-			var aa = chunkify(a.title);
-			var bb = chunkify(b.title);
-
-			for (x = 0; aa[x] && bb[x]; x++) {
-				if (aa[x] !== bb[x]) {
-					var c = Number(aa[x]), d = Number(bb[x]);
-					if (c == aa[x] && d == bb[x]) {
-						return c - d;
-					} else return (aa[x] > bb[x]) ? 1 : -1;
-				}
-			}
-			return aa.length - bb.length;
-		}
-
-        function alphanumBackward(a, b) {
-
-    		var bb = chunkify(a.title);
-			var aa = chunkify(b.title);
-
-			for (x = 0; aa[x] && bb[x]; x++) {
-				if (aa[x] !== bb[x]) {
-					var c = Number(aa[x]), d = Number(bb[x]);
-					if (c == aa[x] && d == bb[x]) {
-						return c - d;
-					} else return (aa[x] > bb[x]) ? 1 : -1;
-				}
-			}
-			return aa.length - bb.length;
-		}
-    },
-
-    // Flatten subPages list
-    subPagesFlatten: function (pages) {
-
-        var output = [];
-
-        process_array(pages);
-
-        return output;
-
-        function process_array(arr) {
-            if(arr.length) {
-                arr.forEach(function(item) {
-                    if (!item) { return; }
-                    process_array(item.subpages || []);
-                    // If only a header for a branch
-                    if (item.url == "") {
-                        return
-                    }
-                    item.subpages =[];
-                    output.push(item);
-                });
-            }
-        }
-    },
-
-    // If ordered is true, the output is an <ol> instead of <ul>
-    //
-    // Special note: If ordered is true, pages whose locale differs from
-    // the current page's locale are omitted, to work around misplaced
-    // localizations showing up in navigation.
-    subPagesFormatList: function (pages, ordered) {
```

Comment:
Nice catch noticing that this isn't used and not including it your new code!

Reply:
Thanks. I tried to minimize the API surface whereever I could.